ingest/src/main/scala/geotrellis/osme/ingest/GenerateHistogram.scala

```diff
@@ -0,0 +1,27 @@
+package geotrellis.osme.ingest
+
+import geotrellis.raster._
+import geotrellis.raster.histogram._
+import geotrellis.spark._
+import geotrellis.spark.io._
+import geotrellis.spark.io.s3._
+import geotrellis.vector._
+import geotrellis.spark.util.SparkUtils
+import org.apache.spark.SparkConf
+import org.apache.spark.rdd.RDD
+
+object GenerateHistogram {
+    def main(args: Array[String]): Unit = {
+        implicit val sc = SparkUtils.createSparkContext("GeoTrellis OSM IngestElevation", new SparkConf(true))
+        try {
+            val reader: FilteringLayerReader[LayerId] = S3LayerReader("my-bucket", "catalog-prefix")
```

Comment:
If you don't force the type here to be `FilteringLayerReader`, but let it be inferred as `S3layerReader` you'll be able to do something like:
```scala
reader.attributeStore.writer(LayerId("NED", 0"), "histogram", histogram)
```